launcher/minecraft/MinecraftInstance.cpp

```diff
@@ -96,8 +96,45 @@
 #include "MangoHud.h"
 #endif
 
+#ifdef WITH_QTDBUS
+#include <QtDBus/QtDBus>
+#endif
+
 #define IBUS "@im=ibus"
 
+static bool switcherooSetupGPU(QProcessEnvironment& env)
+{
+#ifdef WITH_QTDBUS
+    if (QDBusConnection::systemBus().isConnected()) {
```

Comment:
Try to reverse some of the conditions for early returns(`if (!connected()) return false`).
This way you will get rid of the nesting and make the code more readable as suggested in the previous comment.

Reply:
is this a good idea though?

with the code as it is currently it will simply fall back to the same code as if it were compiled without QT Dbus

If I change it to remove the nesting and return prematurely it would result in 2 failure exit locations in the code.

but I'll do it if you insist

Reply:
@TheKodeToad  is that what you want?

Reply:
Perhaps it's confusing to have happy path turn into sad path but it still is a lot nicer IMO

Reply:
I too would prefer less nesting. 

if we need a default behavior that could be placed in in a function and then do something like
```c++
  if (!connected()) return false) {
    return defaultBehavior();
  }
```

Reply:
return false already causes the default behaviour

Reply:
I'll change the code to remove nesting and return early where possible when I get home.